### PR TITLE
two small fixes for python3

### PR DESCRIPTION
**Pull request by fchapoton**
and one tiny fix for speed

### CODE REVIEW

File: src/twisted/mail/relaymanager.py [hunk 1]
```diff
@@ -365,7 +365,7 @@ def hasWaiting(self):
         @return: C{True} if messages are waiting to be relayed.  C{False}
             otherwise.
         """
-        return len(self.waiting) > 0
+        return bool(len(self.waiting))
```
Comment (wiml) [inline src/twisted/mail/relaymanager.py:368]: Out of curiosity, what is the motivation for this change? If the intent is to avoid extra operations, would `bool(self.waiting)` be better?

**Comment by fchapoton** [comment]
Well, I have been told that computing the len only to see if it is not zero
is much slower than using bool, as bool has a smarter and faster way to see
if a list is empty or not.

**Comment by fchapoton** [comment]
oops, it seems that my change is wrong indeed..

**Comment by fchapoton** [comment]
Damn.. I have deleted my fork, and made a new one. But now I am no longer able to change anything to this pull request. Sorry.
